client.go

```diff
@@ -67,7 +119,7 @@ func (c *Client) CreateDownchannel(accessToken string) (<-chan *Message, error)
 
 // Do posts a request to the AVS service's /events endpoint.
 func (c *Client) Do(request *Request) (*Response, error) {
-	body := &bytes.Buffer{}
+	body := NewBlockingReader()
```

Comment:
Thanks for the feedback. I too was slightly dissatisfied with having to roll something custom for this. 

I did start with using an `io.Pipe()` but the problem is that `io.Pipe()` is always unbuffered. To use it, the entire [writing section](https://github.com/rogertalk/go-avs/blob/master/client.go#L72-L101) would need to be in a goroutine to avoid a deadlock before the HTTP body read is called.

An issue similar to this was discussed here: https://groups.google.com/forum/#!topic/golang-dev/k0bSal8eDyE and a buffered pipe was incorporated into [`net/http2/pipe.go`](https://github.com/golang/net/blob/master/http2/pipe.go).

I'm open to suggestions.